### DISCUSSION TITLE
Is there a way to show a input field in the GUI to change a rule used variable?

### DISCUSSION THREAD

**Discussion by manoloxxl** [arendst/Tasmota#17837] [Q&A] [answered]
Hi,

I'm running a Sonoff THR320D with tasmota to control a heater by using the DS18B20 temp sensor. Today everything is controlled by rules and the target temperature is directly in the rule.

For the future i want to build this heater as a user controlable stand alone heater.  

- The gui should be available by connecting to the tasmota AP WLAN (maybe QR Code for the user)
- the user should see a input field to set the wanted temperature (with a valid range that he can use from 5°C up to 40°C)
- the user should see the regular on/off toggle and state
- Nice to have: he should not be able to change the system settings.

Is there a way to get this possible in tasmota? 

Or, i am really don't know as i never evr used it: 
is this possible in a kind of sonoff app and can i flash the device back to the original software?

Best Regards
M

**Comment by barbudor** [comment, marked as answer]
On ESP32 as used by the THR320D, you can use Berry language to extend Tasmota including changing the web UI

Documentation is there: https://tasmota.github.io/docs/Berry/ 
With a cook-book full of examples: https://tasmota.github.io/docs/Berry-Cookbook

The function `web_server.send_content()` should allow you to inject any HTML code in the page and example how to retreive from the action : https://tasmota.github.io/docs/Berry-Cookbook/#adding-a-button-to-the-main-menu

**Reply by manoloxxl** [reply]
Thanks, i will check if this will do the job.

**Comment by danjperron** [comment]
Yes the cookbook method show how to create a button with a constant. 
I took me a while  to figure out on how to do it with a input field ,but I found a way to get input values and send them to the berry script.

Check the file "HydroCycle.be"  in my github: https://github.com/danjperron/microserre/blob/main/HydroCycle.be

![HydroCycleMenu](https://github.com/arendst/Tasmota/assets/4349916/a9ad177c-61e3-43f6-ae4d-228f7743690b)
